### PR TITLE
Prune long long warnings.

### PR DESCRIPTION
**Pull request by mazocomp**
I hope this doesn't happen only at OpenBSD :P

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1055)

### CODE REVIEW

File: auto_tests/file_saving_test.c [hunk 1]
```diff
@@ -85,7 +85,7 @@ static void load_data_decrypted(void)
     uint8_t *cipher = (uint8_t *)malloc(size);
     uint8_t *clear = (uint8_t *)malloc(size - TOX_PASS_ENCRYPTION_EXTRA_LENGTH);
     size_t read_value = fread(cipher, sizeof(*cipher), size, f);
-    printf("Read read_vavue = %u of %ld\n", (unsigned)read_value, size);
+    printf("Read read_vavue = %u of %lld\n", (unsigned)read_value, size);
```
Comment (iphydf) [inline auto_tests/file_saving_test.c]: Cast the argument to `(unsigned long)` instead of using `%lld`.

**Comment by mazocomp** [comment]
@iphydf iirc that'll require changing format string to %uld

**Comment by iphydf** [comment]
Ok, that probably makes sense, because these are "size" type of variables, so they are unsigned.

**Comment by mazocomp** [comment]
@iphydf ok, I'm done.

File: auto_tests/file_saving_test.c [hunk 2]
```diff
@@ -85,7 +85,7 @@ static void load_data_decrypted(void)
     uint8_t *cipher = (uint8_t *)malloc(size);
     uint8_t *clear = (uint8_t *)malloc(size - TOX_PASS_ENCRYPTION_EXTRA_LENGTH);
     size_t read_value = fread(cipher, sizeof(*cipher), size, f);
-    printf("Read read_vavue = %u of %ld\n", (unsigned)read_value, size);
+    printf("Read read_vavue = %u of %lu\n", (unsigned)read_value, (unsigned long) size);
```
Comment (iphydf) [inline auto_tests/file_saving_test.c]: Remove space before cast. More generally, observe the code style around the code you're editing.

**Comment by mazocomp** [comment]
@iphydf doesn't look nice, but done.

**Comment by ibara** [comment]
I am new to c-toxcore, but I have looked over the proposed changes and while I agree with the need to change the format specifiers, I do not believe that the proposed changes as they are now are beneficial to c-toxcore; in fact, there are better solutions for all the proposed changes.

As we learn from https://github.com/TokTok/c-toxcore/blob/master/configure.ac#L270, c-toxcore requires a C99 compiler. The C99 standard includes a format specifier for a size_t, %zu; this is helpful because POSIX specifies only that a size_t be an unsigned integer type, and not the width of the integer. This leads to the common phenomenon where a size_t is 32-bits wide on ILP32 machines and 64-bits wide on LP64 machine, for example. There was previously a discussion about the viability of using the %zu format specifier in c-toxcore, which was ultimately accepted: https://github.com/TokTok/c-toxcore/pull/140, therefore I believe the proposed changed to auto_tests/file_saving_test.c should be amended to use %zu, and perhaps all of c-toxcore should be audited for this idiom, but that's a proposal for a future commit.

As the the changing of time_t, there are some unfortunate POSIX oversights that make this one more difficult. POSIX states only that time_t "shall be integer or real-floating type" (http://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/types.h.html). In other words, it is left up to each operating system to select a type for time_t that they feel is best. In practice, there are only two routinely selected types for time_t, the signed long long used on OpenBSD, NetBSD, and perhaps others, and the signed long used on FreeBSD, Linux, and possibly others as well. However, we cannot rule out the possibility that some POSIX-adhering operating system uses another type, such as a double, for time_t. Using an unsigned long is the worst of all worlds: for all systems, you are needlessly casting; and for ILP32 systems, you still don't solve the 2038 problem.

There are two solutions to this problem:
First is for upstream to use preprocessor defines to select the compiling operating system, which then provides printf lines with the correct format specifiers. Unfortunately, this increases the maintenance burden on upstream, which they may not find a good use of their time. Second is to use the operating system's package building system to keep local patches that use the correct format specifier; such patches are common throughout the OpenBSD ports system, for example. This second solution will result in the rejection of the proposed changes, save for the size_t part, and the changes be maintained elsewhere.

**Comment by mazocomp** [comment]
@ibara wow, you know so lot. Ok, I'll send PR for size_t after taking a rest. time_t is a scary mess.

**Comment by iphydf** [comment]
1. Toxcore requires a somewhat compliant C99 compiler (no VLA support needed, no compound literals, no `inline`, no `restrict`, no flexible array members, no designated initialisers, no static array parameter size decls), and not necessarily a fully compliant standard library. We target the real world that is made of various somewhat compliant C99 compilers and various libcs with a common compliant subset of the C standard library. Thus, while `%zu` has been ruled acceptable, we don't automatically therefore accept all C99 format specifiers or library-defined macros.
2. The values printed here are the result of a subtraction, i.e. the number of seconds spent doing something. Casting that value from whatever it is on any given implementation to something like `unsigned int` makes sense. The values are always positive (unless time flows backwards, which given that these are tests and their success does not depend on time flowing either way we're not terribly worried about), and we only care about an integer number of seconds (so time_t being double poses no issue). We're not dealing with absolute time (relative based on some fixed time), we're dealing with time difference (relative to a very nearby dynamic time).
3. Given that we never deal with absolute time, the 2038 problem is irrelevant.
4. This PR fixes some real and some potential issues.

What are the negative technical consequences of going forward with this?

**Pull request closed without merging**